js/main.js

```diff
@@ -1,21 +1,86 @@
-const getRandomIntInclusive = function (min, max) {
-  // Функция взята с https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/Math/random
-  if (min >= 0 && max >= 0 && min !== max && min < max) {
-    min = Math.ceil(min);
-    max = Math.floor(max);
-    return Math.floor(Math.random() * (max - min + 1)) + min; //Максимум и минимум включаются
-  }
-  return('Введите верные значения диапазона');
+function getRandomPositiveFloat (a, b, digits = 1) {
+  const lower = Math.min(Math.abs(a), Math.abs(b));
+  const upper = Math.max(Math.abs(a), Math.abs(b));
+  const result = Math.random() * (upper - lower) + lower;
+  return result.toFixed(digits);
+}
+
+function getRandomPositiveInteger (a, b) {
+  const lower = Math.ceil(Math.min(Math.abs(a), Math.abs(b)));
+  const upper = Math.floor(Math.max(Math.abs(a), Math.abs(b)));
+  const result = Math.random() * (upper - lower + 1) + lower;
+  return Math.floor(result);
+}
+
+const TYPES_OF_HOUSING = [
+  'palace',
+  'flat',
+  'house',
+  'bungalow',
+  'hotel',
+];
+
+const CHECKIN_TIMES = [
+  '12:00',
+  '13:00',
+  '14:00',
+];
+
+const CHECKOUT_TIMES = [
+  '12:00',
+  '13:00',
+  '14:00',
+];
+
+const ALL_FEATURES = [
+  'wifi',
+  'dishwasher',
+  'parking',
+  'washer',
+  'elevator',
+]
+
+const ALL_PHOTOS = [
+  'https://assets.htmlacademy.ru/content/intensive/javascript-1/keksobooking/duonguyen-8LrGtIxxa4w.jpg',
+  'https://assets.htmlacademy.ru/content/intensive/javascript-1/keksobooking/brandon-hoogenboom-SNxQGWxZQi0.jpg',
+  'https://assets.htmlacademy.ru/content/intensive/javascript-1/keksobooking/claire-rendall-b6kAwr1i0Iw.jpg',
+];
+
+const SIMILAR_COUNT = 10;
+
+const getRandomArrayElement = (elements) => {
+  return elements[getRandomPositiveInteger(0, elements.length - 1)];
 };
 
-getRandomIntInclusive(0, 30);
+// массив случайной длины из неповторяющихся значений. Взял с https://qna.habr.com/q/844269
+const createArr = ([...source], maxLength) => Array.from(
+  { length: Math.min(source.length, 1 + Math.random() * maxLength | 0) },
+  () => source.splice(Math.random() * source.length | 0, 1)[0]
+);
 
-const getRandomDecInclusive = function (min, max, length) {
-  // Функция взята с https://learn.javascript.ru/task/random-min-max
-  if (min >= 0 && max >= 0 && min !== max && min < max) {
-    return (min + Math.random() * (max - min)).toFixed(length);
-  }
-  return('Введите верные значения диапазона');
+const createAd = () => {
+  return {
+    author: {
+      avatar: 'img/avatars/user' + '0' + getRandomPositiveInteger(1, 8) + '.png'
```

Comment:
```suggestion
      avatar: `img/avatars/user0${getRandomPositiveInteger(1, 8)}.png`
```

Почитай про строчные литералы. Давай сразу с ними работать, а не с конкатенацией. 
*хотя знать про неё тоже необходимо